clients/samba/samba.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Immediately abort the script on any error encountered
+set -e
+
+IP_ADDR=$(hostname -i | awk '{print $1}')
+FLAGS=""
+
+if [ "$HIVE_CLIENT_PRIVATE_KEY" != "" ]; then
+    FLAGS="$FLAGS  --unsafe-private-key=0x$HIVE_CLIENT_PRIVATE_KEY"
+fi
+
+ echo $HIVE_CLIENT_PRIVATE_KEY
+
+/opt/samba/bin/samba \
+    --jsonrpc-port=8545 \
+    --jsonrpc-host="${IP_ADDR}" \
+    --p2p-ip="${IP_ADDR}" \
+    --use-default-bootnodes=false \
+    --portal-subnetworks="history-network" \
+    $FLAGS
```

Comment:
```suggestion
#!/bin/bash

# Immediately abort the script on any error encountered
set -e

IP_ADDR=$(hostname -i | awk '{print $1}')
FLAGS=""

if [ "$HIVE_CLIENT_PRIVATE_KEY" != "" ]; then
    FLAGS="$FLAGS  --unsafe-private-key=0x$HIVE_CLIENT_PRIVATE_KEY"
fi

if [ "$HIVE_PORTAL_NETWORKS_SELECTED" != "" ]; then
    FLAGS="$FLAGS  --portal-subnetworks=$HIVE_PORTAL_NETWORKS_SELECTED"
else
    FLAGS="$FLAGS  --portal-subnetworks=history-network"
fi

 echo $HIVE_CLIENT_PRIVATE_KEY

/opt/samba/bin/samba \
    --jsonrpc-port=8545 \
    --jsonrpc-host="${IP_ADDR}" \
    --p2p-ip="${IP_ADDR}" \
    --use-default-bootnodes=false \
    $FLAGS
```

please don't hard code the history network, use the environment variable, I understand samba doesn't support other networks yet

Here is an example
https://github.com/ethereum/hive/blob/34dd8b536ba8b8590028790ee70ffd8c3e0f88a6/clients/trin/trin.sh#L13-L17